extensions/amp-analytics/0.1/linker-manager.js

```diff
@@ -230,26 +230,55 @@ export class LinkerManager {
       return;
     }
 
-    let /** @type {Array} */ domains = config['destinationDomains'];
+    const /** @type {Array} */ domains = config['destinationDomains'];
     // If given domains, but not in the right format.
     if (domains && !Array.isArray(domains)) {
       user().warn(TAG, `${name} destinationDomains must be an array.`);
       return;
     }
 
-    // If no domains given, default to canonical and source.
-    if (!domains) {
-      const {sourceUrl, canonicalUrl} = Services.documentInfoForDoc(
-          this.ampdoc_);
-
-      domains = ([sourceUrl, canonicalUrl])
-          .map(url => urlService.parse(url).hostname);
+    // See if any domains match.
+    if (domains && !domains.includes(hostname)) {
+      return;
     }
 
-    // See if any domains match.
-    if (domains.includes(hostname)) {
-      const newUrl = addParamToUrl(href, name, this.resolvedLinkers_[name]);
-      el.href = newUrl;
+    // If no domains given, default to friendly domain matching.
+    if (!domains) {
+      const {sourceUrl} = Services.documentInfoForDoc(this.ampdoc_);
+      const sourceOrigin = urlService.parse(sourceUrl).hostname;
+      if (!isFriendlyDomains(sourceOrigin, hostname)) {
```

Comment:
what are your thoughts on calling this for both source and canonical domains?

Reply:
+1 should be called for canonical

Reply:
SG

Reply:
done